preprocessing/speaking_status/annotations_v2/analyse_and_plot.py

```diff
@@ -0,0 +1,204 @@
+                        )
+
+    participant_annotations = [
+        participant_annotation[:min_len]
+        for participant_annotation in participant_annotations
+    ]
+    return np.array(prolific_ids), np.array(participant_annotations), total_agreement
+
+
+def main(
+    database_files: list[Path],
+    num_annotated_participants: int,
+    do_plots: bool = True,
+    multithreaded: bool = True,
+    min_aggrement: float = 0.8,
+):
+    all_annotation_data = []
+    for database_file in database_files:
+        all_annotation_data.append(
+            load_json_data(database_file, num_annotated_participants)
+        )
+
+    total_agreement: dict[str, AggrementData] = {}
+    for participant_id in range(1, num_annotated_participants+1):
+        print("\nParticipant", participant_id)
+        prolific_ids, participant_data, total_agreement = get_all_data_for_participant(
+            participant_id, all_annotation_data, total_agreement
+        )
+        vote_data = majority_vote(participant_data, range(len(prolific_ids)))
+        agreement = np.sum(vote_data == participant_data, axis=1) / vote_data.shape[0]
+
+        for annotator, agreement_i in zip(prolific_ids, agreement):
+            print(f"annotator {annotator}, agreement {agreement_i*100:2.2f}%")
+            if annotator[0] not in total_agreement:
+                total_agreement[annotator[0]] = AggrementData([])
+            total_agreement[annotator[0]].data.append(agreement_i)
+
+        participant_data = np.concatenate((participant_data, vote_data[np.newaxis, :]))
+        prolific_ids = np.concatenate((prolific_ids, [["majority_vote"]]))
+
+        if do_plots:
+            plot_annotations(
+                participant_id,
+                prolific_ids,
+                participant_data,
+                multithreaded=multithreaded,
+            )
+
+    print("\nAggregate results")
+    for annotator, agreement in total_agreement.items():
+        agreement.mean = np.mean(agreement.data)
+        agreement.std = np.std(agreement.data)
+        print(
+            f"annotator {annotator}, agreement, mean {agreement.mean*100:2.2f}%, std {agreement.std*100:2.2f}%"
+        )
+
+    print(f"\nProlific annotators that marked participants as missing")
+    for annotator, agreement in total_agreement.items():
+        if agreement.num_marked_missing > 0:
+            print(
+                f"{annotator}, marked missing {agreement.num_marked_missing} participants out of {num_annotated_participants}"
+            )
+
+    max_missing_threshold = 0
+    print(f"\nProlific list over {min_aggrement*100:2.2f}% agreement and missing <= {max_missing_threshold}:")
+    i = 0
+    for annotator, agreement in total_agreement.items():
+        if agreement.mean > min_aggrement and agreement.num_marked_missing <= max_missing_threshold:
+            print(f"{annotator},")
+            i += 1
+    print(f"Total {i} annotators")
+
+
+if __name__ == "__main__":
+    data_files = [
```

Comment:
Please add comments for maintenance: "These json files are obtainable from the covfee admin panel.. blah blah"